### ISSUE TITLE
[FEATURE] Generalize file paths and add options for FUSE path substitution

### ISSUE THREAD

**Issue by renardeinside** [databrickslabs/dbx#50]
## Expected Behavior

Currently, the logic of transforming local paths into dbfs:/ based paths is missing well-defined rules and misses options for /dbfs/ substitutions. 
Would be great to implement the following:
```
if local path is referenced like "file://<>" then simply upload the file and pass dbfs:/ path
if local path is referenced like "file:as-fuse://" then upload the file and pass /dbfs/ path 
```

**Comment by renardeinside** [comment]
This issue needs a bit deeper rework of dbx, we'll include it into 0.1.5.

**Comment by renardeinside** [comment]
Implemented in #81 , will be released in 0.2.0
